climada/engine/uncertainty/unc_cost_benefit.py

```diff
@@ -0,0 +1,227 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define Uncertainty Cost Benefit class
+"""
+
+__all__ = ['UncCostBenefit']
+
+import logging
+import time
+from functools import partial
+import pandas as pd
+
+from climada.engine.uncertainty.base import Uncertainty, UncVar
+from climada.engine.cost_benefit import CostBenefit
+from climada.util import log_level
+
+LOGGER = logging.getLogger(__name__)
+
+# Future planed features:
+# - Add 'efc' (frequency curve) to UncCostBenenfit
+
+class UncCostBenefit(Uncertainty):
+    """
+    Cost Benefit Uncertainty analysis class
+
+    This is the base class to perform uncertainty analysis on the outputs of a
+    climada.engine.costbenefit.CostBenefit() object using the SALib package.
+
+    """
```

Comment:
Can you provide a pseudo-code usage example? I find this really hard to grasp in terms of all the pre-steps that need to be done, and especially highlighting the fact that some measures need to be defined, computed and added as an unc_var to the entity, which in turn is an UncVar class. The measures are no-where explicitly mentioned in the docstrings (yes, even if it's a no-brainer that you need some for doing cost-benefit calcs).

Reply:
I would argue that is what the tutorial is for right?